PersistentObjectException when parsing ConsensusCreateTopic with existing autorenew account

**Detailed Description**
Encountered a transaction in staging that halted mirror node processing

**Actual Behavior**
Steps to reproduce the behavior:
1. Ensure account X exists in DB
2. Parse `ConsensusCreateTopicTransaction` with autoRenewAccount set to X
3. Error:
```
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]: 2020-01-18 04:34:40,012 INFO  [scheduling-3] c.h.m.i.p.r.RecordFileParser Loading version 2 record file: 2020-01-17T22_40_16.328966Z.rcd
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]: 2020-01-18 04:34:40,023 ERROR [scheduling-3] c.h.m.i.p.r.RecordFileParser Exception {}
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]: org.springframework.dao.InvalidDataAccessApiUsageException: detached entity passed to persist: com.hedera.mirror.importer.domain.Entities; nested exception is org.hibernate.PersistentObjectException: detached entity passed to persist: com.hedera.mirror.importer.domain.Entities
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:319) ~[spring-orm-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:255) ~[spring-orm-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:528) ~[spring-orm-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:153) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:178) ~[spring-data-jpa-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:93) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.sun.proxy.$Proxy147.save(Unknown Source) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.GeneratedMethodAccessor74.invoke(Unknown Source) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.cache.interceptor.CacheInterceptor.lambda$invoke$0(CacheInterceptor.java:53) ~[spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.cache.interceptor.CacheAspectSupport.invokeOperation(CacheAspectSupport.java:365) ~[spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:420) ~[spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.cache.interceptor.CacheAspectSupport.execute(CacheAspectSupport.java:345) ~[spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.cache.interceptor.CacheInterceptor.invoke(CacheInterceptor.java:61) ~[spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.sun.proxy.$Proxy147.save(Unknown Source) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.hedera.mirror.importer.parser.record.RecordFileLogger.storeRecord(RecordFileLogger.java:472) ~[classes!/:0.5.1]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.hedera.mirror.importer.parser.record.RecordFileParser.loadRecordFile(RecordFileParser.java:200) [classes!/:0.5.1]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.hedera.mirror.importer.parser.record.RecordFileParser.loadRecordFiles(RecordFileParser.java:291) [classes!/:0.5.1]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:330) [classes!/:0.5.1]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) [spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.lang.Thread.run(Thread.java:834) [?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]: Caused by: org.hibernate.PersistentObjectException: detached entity passed to persist: com.hedera.mirror.importer.domain.Entities
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:127) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:118) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:726) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:694) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.engine.spi.CascadingActions$7.cascade(CascadingActions.java:298) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.engine.internal.Cascade.cascadeToOne(Cascade.java:492) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.engine.internal.Cascade.cascadeAssociation(Cascade.java:416) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.engine.internal.Cascade.cascadeProperty(Cascade.java:218) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.engine.internal.Cascade.cascade(Cascade.java:151) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.AbstractSaveEventListener.cascadeBeforeSave(AbstractSaveEventListener.java:432) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.AbstractSaveEventListener.performSaveOrReplicate(AbstractSaveEventListener.java:269) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.AbstractSaveEventListener.performSave(AbstractSaveEventListener.java:198) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.AbstractSaveEventListener.saveWithGeneratedId(AbstractSaveEventListener.java:128) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.DefaultPersistEventListener.entityIsTransient(DefaultPersistEventListener.java:192) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:135) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.internal.DefaultPersistEventListener.onPersist(DefaultPersistEventListener.java:62) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:108) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.internal.SessionImpl.firePersist(SessionImpl.java:702) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.hibernate.internal.SessionImpl.persist(SessionImpl.java:688) ~[hibernate-core-5.4.9.Final.jar!/:5.4.9.Final]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.orm.jpa.SharedEntityManagerCreator$SharedEntityManagerInvocationHandler.invoke(SharedEntityManagerCreator.java:314) ~[spring-orm-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at com.sun.proxy.$Proxy123.persist(Unknown Source) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.jpa.repository.support.SimpleJpaRepository.save(SimpleJpaRepository.java:554) ~[spring-data-jpa-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.GeneratedMethodAccessor75.invoke(Unknown Source) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.repository.core.support.RepositoryComposition$RepositoryFragments.invoke(RepositoryComposition.java:371) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.repository.core.support.RepositoryComposition.invoke(RepositoryComposition.java:204) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.repository.core.support.RepositoryFactorySupport$ImplementationMethodExecutionInterceptor.invoke(RepositoryFactorySupport.java:657) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.doInvoke(RepositoryFactorySupport.java:621) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.repository.core.support.RepositoryFactorySupport$QueryExecutorMethodInterceptor.invoke(RepositoryFactorySupport.java:605) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:80) ~[spring-data-commons-2.2.3.RELEASE.jar!/:2.2.3.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:366) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:99) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:139) ~[spring-tx-5.2.2.RELEASE.jar!/:5.2.2.RELEASE]
Jan 18 04:34:40 mirrornode-staging1-1 java[23892]:         ... 37 more
```

**Expected Behavior**
Transaction parsed and stored.

**Environment:**
- Java: OpenJDK 11
- OS: Ubuntu 18.04
- Version: v0.5.1

**Additional Context**

--- a/hedera-mirror-coverage/pom.xml
+++ b/hedera-mirror-coverage/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <dependencies>

--- a/hedera-mirror-datagenerator/pom.xml
+++ b/hedera-mirror-datagenerator/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <dependencies>

--- a/hedera-mirror-grpc/pom.xml
+++ b/hedera-mirror-grpc/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <dependencies>

--- a/hedera-mirror-importer/pom.xml
+++ b/hedera-mirror-importer/pom.xml
@@ -13,7 +13,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <properties>

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/Entities.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/Entities.java
@@ -20,7 +20,6 @@ package com.hedera.mirror.importer.domain;
  * ‍
  */
 
-import javax.persistence.CascadeType;
 import javax.persistence.Column;
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
@@ -54,7 +53,7 @@ public class Entities {
     @Column(name = "fk_entity_type_id")
     private Integer entityTypeId;
 
-    @ManyToOne(cascade = CascadeType.ALL)
+    @ManyToOne
     private Entities autoRenewAccount;
 
     private Long autoRenewPeriod;

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/EntityTypeEnum.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/domain/EntityTypeEnum.java
@@ -1,0 +1,36 @@
+package com.hedera.mirror.importer.domain;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
+public enum EntityTypeEnum {
+
+    ACCOUNT(1),
+    CONTRACT(2),
+    FILE(3),
+    TOPIC(4);
+
+    private final int id;
+}

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileLogger.java
@@ -245,6 +245,7 @@ public class RecordFileLogger {
             body = TransactionBody.parseFrom(transaction.getBodyBytes());
         }
 
+        log.trace("Storing transaction body: {}", () -> Utility.printProtoMessage(body));
         long initialBalance = 0;
         Entities entity = null;
         Entities proxyEntity = null;
@@ -469,6 +470,7 @@ public class RecordFileLogger {
             if (proxyEntity != null) {
                 entity.setProxyAccountId(proxyEntity.getId());
             }
+            entity.setAutoRenewAccount(createEntity(entity.getAutoRenewAccount()));
             entity = entityRepository.save(entity);
             sqlInsertTransaction.setLong(F_TRANSACTION.CUD_ENTITY_ID.ordinal(), entity.getId());
         } else {

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/RecordFileParser.java
@@ -197,15 +197,15 @@ public class RecordFileParser implements FileParser {
                                 TransactionRecord txRecord = TransactionRecord.parseFrom(rawBytes);
 
                                 try {
-                                    RecordFileLogger.storeRecord(transaction, txRecord, rawBytes);
-
                                     if (log.isTraceEnabled()) {
                                         log.trace("Transaction = {}, Record = {}", Utility
                                                 .printProtoMessage(transaction), Utility.printProtoMessage(txRecord));
                                     } else {
-                                        log.debug("Stored transaction with consensus timestamp {}", () -> Utility
+                                        log.debug("Storing transaction with consensus timestamp {}", () -> Utility
                                                 .printProtoMessage(txRecord.getConsensusTimestamp()));
                                     }
+
+                                    RecordFileLogger.storeRecord(transaction, txRecord, rawBytes);
                                 } finally {
                                     // TODO: Refactor to not parse TransactionBody twice
                                     DataCase dc = Utility.getTransactionBody(transaction).getDataCase();

--- a/hedera-mirror-importer/src/main/resources/db/migration/V1.17.5__fix_mirror_api_permissions.sql
+++ b/hedera-mirror-importer/src/main/resources/db/migration/V1.17.5__fix_mirror_api_permissions.sql
@@ -1,0 +1,3 @@
+alter default privileges in schema public grant select on tables to ${api-user};
+
+grant select on all tables in schema public to ${api-user};

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileLoggerTopicTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/record/RecordFileLoggerTopicTest.java
@@ -47,6 +47,7 @@ import com.hedera.mirror.importer.KeyConverter;
 import com.hedera.mirror.importer.TestUtils;
 import com.hedera.mirror.importer.TopicIdConverter;
 import com.hedera.mirror.importer.domain.Entities;
+import com.hedera.mirror.importer.domain.EntityTypeEnum;
 import com.hedera.mirror.importer.domain.TopicMessage;
 import com.hedera.mirror.importer.domain.Transaction;
 import com.hedera.mirror.importer.util.Utility;
@@ -54,7 +55,6 @@ import com.hedera.mirror.importer.util.Utility;
 public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
 
     static final String TRANSACTION_MEMO = "transaction memo";
-    static final int TOPIC_ENTITY_TYPE_ID = 4;
     static final String NODE_ID = "0.0.3";
     static final String TRANSACTION_ID = "0.0.9999-123456789";
 
@@ -110,7 +110,33 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
                 .returns("".getBytes(), from(Entities::getSubmitKey))
                 .returns("", from(Entities::getMemo))
                 .returns(false, from(Entities::isDeleted))
-                .returns(TOPIC_ENTITY_TYPE_ID, from(Entities::getEntityTypeId));
+                .returns(EntityTypeEnum.TOPIC.getId(), from(Entities::getEntityTypeId));
+    }
+
+    // https://github.com/hashgraph/hedera-mirror-node/issues/501
+    @Test
+    void createTopicTestExistingAutoRenewAccount() throws Exception {
+        Entities autoRenewAccount = createEntity(100L, EntityTypeEnum.ACCOUNT);
+        var topicId = 200L;
+        var consensusTimestamp = 2_000_000L;
+        var responseCode = ResponseCodeEnum.SUCCESS;
+        var transaction = createCreateTopicTransaction(null, null, null, autoRenewAccount.getEntityNum(), null);
+        var transactionRecord = createTransactionRecord(TopicID.newBuilder().setTopicNum(topicId)
+                .build(), null, null, consensusTimestamp, responseCode);
+
+        RecordFileLogger.storeRecord(transaction, transactionRecord);
+        RecordFileLogger.completeFile("", "");
+
+        var entity = entityRepository.findByPrimaryKey(0L, 0L, topicId).get();
+        assertTransactionInRepository(responseCode, consensusTimestamp, entity.getId());
+        assertEquals(4L, entityRepository.count()); // Node, payer, topic, autorenew
+        assertThat(entity)
+                .returns("".getBytes(), from(Entities::getKey))
+                .returns("".getBytes(), from(Entities::getSubmitKey))
+                .returns("", from(Entities::getMemo))
+                .returns(false, from(Entities::isDeleted))
+                .returns(EntityTypeEnum.TOPIC.getId(), from(Entities::getEntityTypeId))
+                .returns(autoRenewAccount, from(Entities::getAutoRenewAccount));
     }
 
     @Test
@@ -160,6 +186,9 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
         // Store topic to be updated.
         var topic = createTopicEntity(topicId, expirationTimeSeconds, expirationTimeNanos, adminKey, submitKey, memo,
                 autoRenewAccount, autoRenewPeriod);
+        if (topic.getAutoRenewAccount() != null) {
+            topic.setAutoRenewAccount(entityRepository.save(topic.getAutoRenewAccount()));
+        }
         entityRepository.save(topic);
 
         var responseCode = ResponseCodeEnum.SUCCESS;
@@ -265,6 +294,9 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
         // Store topic to be updated.
         var topic = createTopicEntity(topicId, expirationTimeSeconds, expirationTimeNanos, adminKey, submitKey, memo,
                 autoRenewAccount, autoRenewPeriod);
+        if (topic.getAutoRenewAccount() != null) {
+            topic.setAutoRenewAccount(entityRepository.save(topic.getAutoRenewAccount()));
+        }
         entityRepository.save(topic);
 
         // Setup the expected entity.
@@ -561,6 +593,15 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
                 .setMemo(TRANSACTION_MEMO);
     }
 
+    private Entities createEntity(long num, EntityTypeEnum entityType) {
+        Entities entities = new Entities();
+        entities.setEntityShard(0L);
+        entities.setEntityRealm(0L);
+        entities.setEntityNum(num);
+        entities.setEntityTypeId(entityType.getId());
+        return entityRepository.save(entities);
+    }
+
     private Entities createTopicEntity(TopicID topicId, Long expirationTimeSeconds, Integer expirationTimeNanos,
                                        Key adminKey, Key submitKey, String memo, Long autoRenewAccount,
                                        Long autoRenewPeriod) {
@@ -573,7 +614,7 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
             autoRenewEntity.setEntityShard(topic.getEntityShard());
             autoRenewEntity.setEntityRealm(topic.getEntityRealm());
             autoRenewEntity.setEntityNum(autoRenewAccount);
-            autoRenewEntity.setEntityTypeId(1);
+            autoRenewEntity.setEntityTypeId(EntityTypeEnum.ACCOUNT.getId());
             topic.setAutoRenewAccount(autoRenewEntity);
         }
         if (autoRenewPeriod != null) {
@@ -589,7 +630,7 @@ public class RecordFileLoggerTopicTest extends AbstractRecordFileLoggerTest {
             topic.setSubmitKey(submitKey.toByteArray());
         }
         topic.setMemo(memo);
-        topic.setEntityTypeId(TOPIC_ENTITY_TYPE_ID);
+        topic.setEntityTypeId(EntityTypeEnum.TOPIC.getId());
         return topic;
     }
 

--- a/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/EntityRepositoryTest.java
+++ b/hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/repository/EntityRepositoryTest.java
@@ -37,6 +37,7 @@ public class EntityRepositoryTest extends AbstractRepositoryTest {
         autoRenewAccount.setEntityShard(0L);
         autoRenewAccount.setEntityRealm(0L);
         autoRenewAccount.setEntityNum(101L);
+        autoRenewAccount = entityRepository.save(autoRenewAccount);
 
         Entities proxyEntity = new Entities();
         proxyEntity.setEntityTypeId(entityTypeId);
@@ -59,11 +60,6 @@ public class EntityRepositoryTest extends AbstractRepositoryTest {
         entity.setProxyAccountId(proxyEntity.getId());
         entity.setSubmitKey("SubmitKey".getBytes());
         entity = entityRepository.save(entity);
-
-        assertThat(entityRepository.findByPrimaryKey(autoRenewAccount.getEntityShard(),
-                autoRenewAccount.getEntityRealm(), autoRenewAccount.getEntityNum()))
-                .get()
-                .isEqualToIgnoringGivenFields(autoRenewAccount, "id");
 
         assertThat(entityRepository
                 .findByPrimaryKey(entity.getEntityShard(), entity.getEntityRealm(), entity.getEntityNum()))

--- a/hedera-mirror-protobuf/pom.xml
+++ b/hedera-mirror-protobuf/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <properties>

--- a/hedera-mirror-rest/package.json
+++ b/hedera-mirror-rest/package.json
@@ -1,6 +1,6 @@
 {
   "name": "hedera-mirror-rest",
-  "version": "0.5.1",
+  "version": "0.5.2",
   "description": "Hedera Mirror Node REST API",
   "main": "server.js",
   "scripts": {

--- a/hedera-mirror-rest/pom.xml
+++ b/hedera-mirror-rest/pom.xml
@@ -11,7 +11,7 @@
     <parent>
         <groupId>com.hedera</groupId>
         <artifactId>hedera-mirror-node</artifactId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <build>

--- a/hedera-mirror-test/pom.xml
+++ b/hedera-mirror-test/pom.xml
@@ -12,7 +12,7 @@
     <parent>
         <artifactId>hedera-mirror-node</artifactId>
         <groupId>com.hedera</groupId>
-        <version>0.5.1</version>
+        <version>0.5.2</version>
     </parent>
 
     <properties>

--- a/pom.xml
+++ b/pom.xml
@@ -4,7 +4,7 @@
 
     <groupId>com.hedera</groupId>
     <artifactId>hedera-mirror-node</artifactId>
-    <version>0.5.1</version>
+    <version>0.5.2</version>
     <description>Hedera Mirror Node mirrors data from Hedera nodes and serves it via an API</description>
     <inceptionYear>2019</inceptionYear>
     <modelVersion>4.0.0</modelVersion>

